models/articles.model.js

```diff
@@ -0,0 +1,33 @@
+const db = require('../db/connection')
+
+exports.fetchArticleById = (id) => {
+    const query = `SELECT * FROM articles WHERE article_id = $1;`
+    return db.query(query, [id]).then(({rows}) => {
+        if (rows.length === 0){
+            return Promise.reject({status: 404, msg: 'Not Found'})
+        }
+        return rows[0]
+    })
+}
+
+exports.fetchArticles = () => {
+    const query = `SELECT * FROM articles ORDER BY created_at DESC`
+    return db.query(query).then(({rows}) => {
+        const newArr = rows.map(article => {
+            let counter = 0
+            delete article.body
+            return db.query(`SELECT * FROM comments`).then(res => {
+                const comRow = res.rows
+                comRow.forEach(com => {
+                    if (com.article_id === article.article_id){
+                        counter++
+                    }
+                })
+                article.comment_count = counter
+                return article
+            })
+        })
+        
+        return Promise.all(newArr)
+    })
+}
```

Comment:
So this works, and the test will prove that.

However having logic in here with a select query to the comments table isn't needed.

Instead, maybe you could `JOIN` on the `comments` table to the first `SQL` statement, select the columns you want, and use an sql tool which will allow you to `count` all comments (look at aggregate functions, here's our nc notes on them: https://notes.northcoders.com/courses/js-back-end/sql-relationships#group-by-and-aggregate-functions). So in summary this could, and probably should be done in 1 sql query.